### PR TITLE
[feat]: [CDS-72114]: Adding overrides v2 implementation for app settings and connection strings

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Adding overrides v2 implementation for app settings and connection strings

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49236)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger ss

**Comment by markoharness** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java` line 386 at r1](https://reviewable.io/reviews/harness/harness-core/49236#-NXu3JfI0bv8TO5cTLQe:-NXu3JfI0bv8TO5cTLQf:beo404z) ([raw file](https://github.com/harness/harness-core/blob/5561cc9b4e81f2f93ed002baa1f607a01d3fa5ec/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java#L386)):*
> ```Java
>     for (ServiceOverridesType overridesType : OVERRIDE_IN_REVERSE_PRIORITY) {
>       if (overrideV2Configs.containsKey(overridesType)) {
>         overrideV2Configs.get(overridesType);
> ```

why is this line necessary?

**Comment by markoharness** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java` line 421 at r1](https://reviewable.io/reviews/harness/harness-core/49236#-NXu445gDMc7Pl5tpMXV:-NXu445i6JEg-SLomEQ1:b2fot6y) ([raw file](https://github.com/harness/harness-core/blob/5561cc9b4e81f2f93ed002baa1f607a01d3fa5ec/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java#L421)):*
> ```Java
>     for (ServiceOverridesType overridesType : OVERRIDE_IN_REVERSE_PRIORITY) {
>       if (overrideV2Configs.containsKey(overridesType)) {
>         overrideV2Configs.get(overridesType);
> ```

same here? Is this line necessary?

**Review by markoharness — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49236)* status: 2 file(s) to review. 0 of 2 approvals obtained.

**Comment by lovish1999** [comment]
> _[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java` line 421 at r1](https://reviewable.io/reviews/harness/harness-core/49236#-NXu445gDMc7Pl5tpMXV:-NXu445i6JEg-SLomEQ1:b2fot6y) ([raw file](https://github.com/harness/harness-core/blob/5561cc9b4e81f2f93ed002baa1f607a01d3fa5ec/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java#L421)):_
> 
> > ```java
> >     for (ServiceOverridesType overridesType : OVERRIDE_IN_REVERSE_PRIORITY) {
> >       if (overrideV2Configs.containsKey(overridesType)) {
> >         overrideV2Configs.get(overridesType);
> > ```
> 
> same here? Is this line necessary?

No , removed them.

**Comment by lovish1999** [comment]
trigger uts

**Review by markoharness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49236)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java [hunk 1]
```diff
@@ -360,6 +361,73 @@ && isNotEmpty(overrideV2Configs.get(overridesType).getSpec().getConfigFiles()))
...
+  private List<ApplicationSettingsConfiguration> prepareFinalAppSettingsV2(
+      NGServiceV2InfoConfig serviceV2Config, Map<ServiceOverridesType, NGServiceOverrideConfigV2> overrideV2Configs) {
+    List<ApplicationSettingsConfiguration> applicationSettingsConfiguration = getSvcAppSettings(serviceV2Config);
+
+    for (ServiceOverridesType overridesType : OVERRIDE_IN_REVERSE_PRIORITY) {
+      if (overrideV2Configs.containsKey(overridesType)) {
+        ServiceOverridesSpec spec = overrideV2Configs.get(overridesType).getSpec();
+        ApplicationSettingsConfiguration configuration = spec.getApplicationSettings();
+        if (configuration != null) {
+          applicationSettingsConfiguration = Collections.singletonList(configuration);
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java:389]: Can we just declare "configuration" variable out of for loop. And set applicationSettingsConfiguration out side of for loop. Not needed to convert to list in every iteration

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Comment by tchaurasiya** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java` line 389 at r2](https://reviewable.io/reviews/harness/harness-core/49236#gh-1234137054:-NYMLmkkEm1B3EcsBmIA:b-trh6o) ([raw file](https://github.com/harness/harness-core/blob/f0fd75f3a9fb8f7c235240322186abcdb2f6d9e2/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepOverrideHelper.java#L389)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Can we just declare "configuration" variable out of for loop. And set applicationSettingsConfiguration out side of for loop. Not needed to convert to list in every iteration
</blockquote></details>

Discussed offline, Will take it up in next pr

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49236)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions.

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49236)* status: Some of the checks are too old. 1. unresolved discussions.

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**